--- openl2tp-1.8/Makefile~	2016-10-22 19:35:05.000000000 +0200
+++ openl2tp-1.8/Makefile	2016-10-22 19:39:02.267356788 +0200
@@ -29,28 +29,6 @@
 # Build for UML environment?
 # UML_TARGET=		y
 
-# Location of shared libs. This is usually /usr/lib, but can be
-# /usr/lib32 or /usr/lib64 on 64-bit multi-arch systems. On 64-bit
-# systems which don't have 32-bit multi-arch libs installed, the
-# standard says build with /usr/lib. But Fedora uses /usr/lib64. We
-# try to handle these cases by deriving the CPU architecture
-# name. Things are made more complicated by the fact that the CPU
-# architecture is output by either uname -m or uname -p, depending on
-# uname version. Override SYS_LIBDIR from the command line when
-# necessary.
-SYS_LIBDIR=/usr/lib
-ifeq ($(CROSS_COMPILE),)
- ifeq ($(strip $(wildcard /etc/debian_version*)),)
-  ifeq ($(shell uname -p),x86_64)
-  SYS_LIBDIR=/usr/lib64
-  else
-   ifeq ($(shell uname -m),x86_64)
-   SYS_LIBDIR=/usr/lib64
-   endif
-  endif
- endif
-endif
-
 # Points to pppd install.  By default, pppd headers are assumed to be
 # in the pppd subdirectory of the compiler's default search path
 # (e.g. /usr/include/pppd). but can be pointed to another directory if
@@ -62,7 +62,7 @@
 
 export PPPD_VERSION PPPD_SRCDIR PPPD_LIBDIR READLINE_LDFLAGS READLINE_CFLAGS
 export CROSS_COMPILE AS LD CC AR NM STRIP OBJCOPY OBJDUMP INSTALL UML_TARGET
-export DESTDIR SYS_LIBDIR
+export DESTDIR
 
 # Feature options are exported to sub-makes
 export L2TP_FEATURE_LAC_SUPPORT
